src/connection/server_support.c

```diff
@@ -2329,24 +2360,22 @@ css_change_ha_server_state (THREAD_ENTRY * thread_p, HA_SERVER_STATE state, bool
 	{
 	  break;
 	}
+      if (!HA_DISABLED () && state == HA_SERVER_STATE_TO_BE_ACTIVE)
+	{
+	  cubreplication::replication_node_manager::commute_to_master_state ([thread_p] ()
```

Comment:
We have to be sure that another css_change_ha_server_state request (on current thread) is executed after commute job (executed in another thread). Just to be sure that we do not change the old logic.

Reply:
I've checked and the cases when a concurrent css_change_ha_server_state call is made (during a commute done on another case) is safe since the TO_BE_ACTIVE/STANDBY states forbit transitions to other states besides TO_BE_ACTIVE -> ACTIVE & TO_BE_STANDBY -> STANDBY (as expressed by ha_Server_state_transition).